### PR TITLE
Add support for kubectl create quota command

### PR DESCRIPTION
**Pull request by sttts**
Follow-up of https://github.com/kubernetes/kubernetes/pull/19625

```
Create a resourcequota with the specified name, hard limits and optional scopes

Usage:
  kubectl create quota NAME [--hard=key1=value1,key2=value2] [--scopes=Scope1,Scope2] [--dry-run=bool] [flags]

Aliases:
  quota, q


Examples:
  // Create a new resourcequota named my-quota
  $ kubectl create quota my-quota --hard=cpu=1,memory=1G,pods=2,services=3,replicationcontrollers=2,resourcequotas=1,secrets=5,persistentvolumeclaims=10

  // Create a new resourcequota named best-effort
  $ kubectl create quota best-effort --hard=pods=100 --scopes=BestEffort
```

### CODE REVIEW

File: pkg/kubectl/cmd/create_quota.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
```
Comment (derekwaynecarr) [inline pkg/kubectl/cmd/create_quota.go]: Fix year

File: pkg/kubectl/cmd/create_quota.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		Run: func(cmd *cobra.Command, args []string) {
+			err := CreateQuota(f, cmdOut, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddGeneratorFlags(cmd, cmdutil.ResourceQuotaV1GeneratorName)
+	cmd.Flags().String("hard", "", "A comma-delimited set of resource=quantity pairs that define a hard limit.")
```
Comment (derekwaynecarr) [inline pkg/kubectl/cmd/create_quota.go]: I would like to support scopes

**Comment by derekwaynecarr** [comment]
Didn't do a deep review but please add support for specifying scopes as well via a flag

**Comment by sttts** [comment]
@derekwaynecarr scopes are coming... only did a quick rebase and some fixes for your old review comments.

**Comment by sttts** [comment]
@derekwaynecarr ptal, scopes are done.

**Comment by sttts** [comment]
@maazkhan can you make the cla bot happy with a short confirmation that you are ok with the commit?

**Comment by maazkhan** [comment]
ok to commit.

**Comment by derekwaynecarr** [comment]
@eparis - what is needed so @sttts is automatically cla:yes ?

**Comment by sttts** [comment]
@derekchiang I get automatic cla:yes. It's because I rebased the first commit by @maazkhan.

**Comment by sttts** [comment]
@derekwaynecarr scopes are done. Can you take another look?

File: test/e2e/kubectl.go [hunk 3]
```diff
@@ -1149,6 +1150,74 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
...
+				framework.Failf("Failed getting quota %s: %v", quotaName, err)
+			}
+
+			if len(quota.Spec.Scopes) != 0 {
+				framework.Failf("Expected empty scopes, got %v", quota.Spec.Scopes)
+			}
+			if len(quota.Spec.Hard) != 2 {
+				framework.Failf("Expected two resources, got %v", quota.Spec.Hard)
+			}
+			if r, found := quota.Spec.Hard[api.ResourcePods]; !found || r.MilliValue() != 1000000*1000 {
```
Comment (derekwaynecarr) [inline test/e2e/kubectl.go]: nit: the check is cleaner to me if we just have a 

```
if expected := resource.MustParse("100000); r, found := ...; !found || r.Cmp(expected) != 0 {
...
}
```

File: test/e2e/kubectl.go [hunk 4]
```diff
@@ -1149,6 +1150,74 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
...
+				framework.Failf("Expected pods=1000000, got %v", r)
+			}
+			if r, found := quota.Spec.Hard[api.ResourceServices]; !found || r.MilliValue() != 1000000*1000 {
```
Comment (derekwaynecarr) [inline test/e2e/kubectl.go]: same nit here as above.

**Comment by derekwaynecarr** [comment]
a couple nits, squash commtis, and then this looks ok to me.

File: test/e2e/kubectl.go (hunk 3 above)
Comment (sttts) [inline test/e2e/kubectl.go]: Go only allow one simple statements. We have to write it in two lines:
```go
r, found := quota.Spec.Hard[api.ResourcePods]
			if expected := resource.MustParse("1000000"); !found || (&r).Cmp(expected) != 0 {
```

**Comment by deads2k** [comment]
@ncdc Make lawyer-bot happy?

@sttts Mind squashing in the generation. Otherwise we have a commit history where some commits are invalid and that's not much generated code.

**Comment by sttts** [comment]
@deads2k the cla label hopefully is fine with the human-approved override. I have squashed the generated files into the previous patch.

**Comment by deads2k** [comment]
> @deads2k the cla label hopefully is fine with the human-approved override. I have squashed the generated files into the previous patch.

Yeah, but I hate chasing them around.  It'll be like @kargakis all over again!

**Comment by kargakis** [comment]
Let's hope we will be luckier with the next bot: https://github.com/kubernetes/kubernetes/issues/27796

**Pull request merged**